thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/cache/CollectionMaxDataTimeCacheLoader.java

```diff
@@ -1,85 +1,69 @@
+  private final Map<String, Long> collectionToPrevMaxDataTimeMap = new ConcurrentHashMap<String, Long>();
 
-   zkUrl = pinotThirdEyeClientConfig.getZookeeperUrl();
-   clusterName = pinotThirdEyeClientConfig.getClusterName();
-
-   ZkClient zkClient = new ZkClient(zkUrl);
-   zkClient.setZkSerializer(new ZNRecordSerializer());
-   zkClient.waitUntilConnected();
-
-   ZkBaseDataAccessor<ZNRecord> baseAccessor = new ZkBaseDataAccessor<>(zkClient);
-   String propertyStorePath = PropertyPathConfig.getPath(PropertyType.PROPERTYSTORE, clusterName);
-
-   propertyStore = new ZkHelixPropertyStore<ZNRecord>(baseAccessor, propertyStorePath, null);
+  public CollectionMaxDataTimeCacheLoader(PinotThirdEyeClientConfig pinotThirdEyeClientConfig,
+      LoadingCache<String, CollectionSchema> collectionSchemaCache, LoadingCache<PinotQuery, ResultSetGroup> resultSetGroupCache) {
+    this.collectionSchemaCache = collectionSchemaCache;
+    this.resultSetGroupCache = resultSetGroupCache;
   }
 
   @Override
   public Long load(String collection) throws Exception {
-
     LOGGER.info("Loading maxDataTime cache {}", collection);
-
-    int options = AccessOption.PERSISTENT;
-    String offlineTablePropertyStorePath = "/SEGMENTS/" + collection + "_OFFLINE";
-    String realtimeTablePropertyStorePath = "/SEGMENTS/" + collection + "_REALTIME";
-
     long maxTime = 0;
-    List<Stat> stats = new ArrayList<>();
-    List<ZNRecord> zkSegmentDataList = new ArrayList<>();;
-    if (propertyStore != null) {
-      if (propertyStore.exists(realtimeTablePropertyStorePath, options)) {
-        zkSegmentDataList.addAll(propertyStore.getChildren(realtimeTablePropertyStorePath, stats, options));
-      }
-      if (propertyStore.exists(offlineTablePropertyStorePath, options)) {
-        zkSegmentDataList.addAll(propertyStore.getChildren(offlineTablePropertyStorePath, stats, options));
-      }
-      if (zkSegmentDataList != null && !zkSegmentDataList.isEmpty()) {
-        for (ZNRecord record : zkSegmentDataList) {
-          try {
-            long endTime = Long.parseLong(record.getSimpleField("segment.end.time"));
-            TimeUnit timeUnit = TimeUnit.valueOf(record.getSimpleField("segment.time.unit"));
-            long endTimeMillis = timeUnit.toMillis(endTime);
-            if (endTimeMillis > maxTime) {
-              maxTime = endTimeMillis;
-            }
-          } catch (Exception e) {
-            LOGGER.warn("Exception parsing metadata:{}", record, e);
-          }
-        }
+    try {
+      TimeSpec timeSpec = collectionSchemaCache.get(collection).getTime();
+      long prevMaxDataTime = getPrevMaxDataTime(collection, timeSpec);
+      String maxTimePql = String.format(COLLECTION_MAX_TIME_QUERY_TEMPLATE, timeSpec.getColumnName(), collection, timeSpec.getColumnName(),
+          prevMaxDataTime);
+      PinotQuery maxTimePinotQuery = new PinotQuery(maxTimePql, collection);
+      ResultSetGroup resultSetGroup = resultSetGroupCache.get(maxTimePinotQuery);
```

Comment:
call refresh on resultSetGroupCache before calling get. We dont want to use cache for the maxTimePinotQuery, only way to enforce that is to call refresh that will re-run the query